### DISCUSSION TITLE
use with typescript

### DISCUSSION THREAD

**Discussion by joprice** [nojaf/vite-plugin-fable#9] [General]
Is there a way to use the plugin in a typescript project? I currently get `Cannot find module './Util.fs' or its corresponding type declarations.`, since the extension is unknown to typescript. Perhaps fable's typescript output could be used to avoid having to define typings for every file?

**Comment by nojaf** [comment]
Hello,

TypeScript is currently not supported by [Fable.Compiler](https://www.nuget.org/packages/Fable.Compiler/4.0.0-alpha-008), so there is no way to configure anything.

I'm not quite sure what your exact use case is, but if you need `.d.ts` files to exist on disk this plugin cannot do this.

**Comment by joprice** [comment]
I ended up using the standard approach from here https://fable.io/docs/getting-started/typescript.html which is working fine. I’m not familiar with the internals. I assumed this functionality would be available programmatically if the cli supports it.

**Reply by nojaf** [reply]
You assumed too much, this is not using the cli at all, kinda the whole point of the plugin.

**Reply by joprice** [reply]
I didn't mean that it called the cli, but that there's some common internal code that both the cli and this lib use that can receive a language enum to generate typescript (like `--lang ts`). Since then I've found that typescript output is not always reliable at the moment, and I rewrote the code that was consuming the output of my fable command, so I may give this another try.
